Gifski/GifskiWrapper.swift

```diff
@@ -0,0 +1,84 @@
+import Foundation
+
+enum GifskiWrapperError: UInt32, Error {
+    /** one of input arguments was NULL */
+    case nullArg = 1
+    /** a one-time function was called twice, or functions were called in wrong order */
+    case invalidState
+    /** internal error related to palette quantization */
+    case quant
+    /** internal error related to gif composing */
+    case gif
+    /** internal error related to multithreading */
+    case threadLost
+    /** I/O error: file or directory not found */
+    case notFound
+    /** I/O error: permission denied */
+    case permissionDenied
+    /** I/O error: file already exists */
+    case alreadyExists
+    /** invalid arguments passed to function */
+    case invalidInput
+    /** misc I/O error */
+    case timedOut
+    /** misc I/O error */
+    case writeZero
+    /** misc I/O error */
+    case interrupted
+    /** misc I/O error */
+    case unexpectedEof
+    /** progress callback returned 0, writing aborted */
+    case aborted
+    /** should not happen, file a bug */
+    case other
+}
+
+final class GifskiWrapper {
+    private let pointer: OpaquePointer
+
+    init?(settings: GifskiSettings) {
+        var settings = settings
+        guard let pointer = gifski_new(&settings) else {
+            return nil
+        }
+        self.pointer = pointer
+    }
+
+    deinit {
+        gifski_drop(pointer)
+    }
+
+    func setProgressCallback(context: UnsafeMutableRawPointer, cb: @escaping (@convention(c) (UnsafeMutableRawPointer?) -> Int32)) {
+        gifski_set_progress_callback(pointer, cb, context)
+    }
+
+    func addFrameARGB(index: Int, image: CGImage, fps: Double) throws {
+        let buffer = CFDataGetBytePtr(image.dataProvider!.data)
```

Comment:
Just a thought, maybe we should not force-unwrap `dataProvider`? I know I'm the one that did it, but in hindsight, I guess we can't be sure it won't be `nil`, right?

Reply:
Yes I guess that would be best. Should probably unwrap at call site and change the method signature to take CFData instead of the image then?

Reply:
Yes